### ISSUE TITLE
[clang-tidy] `misc-const-correctness` false positive

### ISSUE THREAD

**Issue by davidmenggx** [llvm/llvm-project#216048]
The following suggestion does not compile:
```cpp
void func(int* p) {
    int* r = p;
    int* v = p;
    *(r = v) = 0;
}
```
```
<source>:2:5: warning: pointee of variable 'r' of type 'int *' can be declared 'const' [misc-const-correctness]
    2 |     int* r = p;
      |     ^  
      |         const
```
Godbolt (trunk): https://godbolt.org/z/3o8qnEKs3
